### DISCUSSION TITLE
Accessing currently hovered Select Dropdown item

### DISCUSSION THREAD

**Discussion by ruzih** [mantinedev/mantine#3642] [Q&A]
Hello,

I'm currently building out some custom components with Mantine, and would like to display the currently hovered item (via mouse and keyboard nav) onto the Select field. Is there a way to access the SelectItem's `hovered` prop or `onItemHover` callback so that I can grab the data label as they are hovered? Or perhaps another solution to achieve it?

Thanks in advance.

**Comment by rtivital** [comment]
I advise to build a custom component with Input and Popover for this

**Reply by ruzih** [reply]
I see. Is there not a way to grab a ref of the dropdown elements?

**Comment by ruzih** [comment]
Followup:
I'm in the process of implementing a custom Popover dropdown select list, with a `Stack` of `Button` Mantine components for the list. I see that there's a `use-hover` hook to detect element hovering, but it doesn't appear to be working at all like the documentation. Any insight?

```
  const renderDropdownItems = () => options.map((data: SelectItem, index: number) =>  {
    const { hovered, ref } = useHover();
    const { value, label } = data;
    return (
      <Button
        onClick={() => onItemSelect(value, label)}
        ref={ref}
        component="div"
      >
          {hovered ? label : value}
      </Button>
    );
  })
```
